securedrop/journalist_app/main.py

```diff
@@ -112,11 +123,13 @@ def reply():
              config.JOURNALIST_KEY],
             output=current_app.storage.path(g.filesystem_id, filename),
         )
-        reply = Reply(g.user, g.source, filename)
-
         try:
+            reply = Reply(g.user, g.source, filename)
             db.session.add(reply)
             db.session.commit()
+            seen_reply = SeenReply(reply_id=reply.id, journalist_id=g.user.id)
+            db.session.add(seen_reply)
+            db.session.commit()
```

Comment:
It might be better to make the `commit` immediately before the `SeenReply` is constructed a `flush` instead, so if there's  a problem creating the `SeenReply`, the reply won't have been committed.